src/Features/Blockcore.Features.NodeHost/Authentication/AppSettingsGetApiKeyQuery.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Blockcore.Features.NodeHost.Settings;
+using Microsoft.Extensions.Options;
+
+namespace Blockcore.Features.NodeHost.Authentication
+{
+    public class AppSettingsGetApiKeyQuery : IGetApiKeyQuery
+    {
+        private BlockcoreSettings settings;
```

Comment:
Shouldn't this be the nodehostsettings?

Reply:
nodehostsettings already exists and is a ini-file based configuration. This is a complex type that is serialized from the appsettings.json file.